packages/codegen/generate.test.utils.js

```diff
@@ -39,21 +58,21 @@ function _renderDeveloperFriendlyErrorMessage (data, errors, context) {
     }
 
     if (get(context, 'variables')) {
-        msg.push(`VARIABLES: ${JSON.stringify(context.variables)}`)
+        msg.push(`VARIABLES: ${JSON.stringify(context.variables, getCycleReplacer())}`)
```

Comment:
it looks strange. You don't have a case for cycles inside context.variables

Reply:
errors have cyclick links in some cases